### ISSUE TITLE
Support for Ruby 2.2

### ISSUE THREAD

**Issue by ellingtonjp** [ruby-next/ruby-next#40]
> Please, open an issue if you would like us to support older Ruby versions.

First of all, thanks for the hard work everyone! I'm super excited for this project. Unfortunately I'm stuck with Ruby 2.2 (MRI) at work. Would love to see support for our very old version.

Any plans to support rubies as far back as 2.2?

**Comment by palkan** [comment]
> Any plans to support rubies as far back as 2.2?

If there is a demand for that then we should try to do that.

Personally, I'm not a big fan of supported "dead" Rubies, since it could slow down the evolution of the language. On the other hand, I understand that things happen, and if Ruby Next could make you happier we should try to do that 🙂

As far as I understand, you want to use Ruby Next for application development, not gem development, right?
In that case, we need only transpiling part (APIs for such old Rubies couldn't be backported via refinements efficiently, so we have to use monkey patches, and for that we can rely on [`backports` gem](https://github.com/marcandre/backports) since it supports very old Rubies).

And we need runtime transpiling, I guess. It's better to do that with the help of Bootsnap (which does `require`-patching for us), which stopped supporting <2.3 in its 1.4+. We can use [1.3.x](https://github.com/Shopify/bootsnap/blob/v1.3.2/bootsnap.gemspec); just need to make sure Ruby Next is compatible with this version.

Finally, the Ruby Next code itself uses syntax (e.g., `&.`) and APIs from 2.5+. Thus, we need to transpile Ruby Next itself! That's actually in my backlog.

To sum up, sounds like it's doable, though would require some work.

**Comment by ellingtonjp** [comment]
I totally understand not wanting to support rubies this old. We are stuck in the stone ages!

> As far as I understand, you want to use Ruby Next for application development, not gem development, right?
Yep! We aren't currently doing any gem development.

> Thus, we need to transpile Ruby Next itself!
Nextception? :P 

Thanks for the response. I will be watching this project closely! We have other parts of our system that are currently supported, so I'll be looking at trying ruby-next for those parts.

**Comment by palkan** [comment]
Hey @ellingtonjp!

Ruby 2.2 support has been added. It will be released in the next version (planned on Sep 2nd).

It's no possible to use the master version from GitHub, 'cause we do not store the transpiled source code in Git (it's only included into the `*.gem` bundle). And we cannot self-transpile Ruby Next on Ruby 2.2)
